jmclient/jmclient/blockchaininterface.py

```diff
@@ -406,20 +406,27 @@ def query_utxo_set(self, txout, includeconf=False, includeunconf=False):
         return result
 
     def estimate_fee_per_kb(self, N):
+        """ The argument N may be either a number of blocks target,
+        for estimation of feerate by Core, or a number of satoshis
+        per kilo-vbyte (see `fee_per_kb_has_been_manually_set` for
+        how this is distinguished).
+        In the latter case, it is prevented from falling below the
+        minimum allowed feerate for relay on the Bitcoin network.
+        """
         if super().fee_per_kb_has_been_manually_set(N):
-            # use the local bitcoin core relay fee as floor to avoid relay problems
+            # use the local bitcoin core relay fee as floor to avoid relay
+            # problems
             btc_relayfee = -1
             rpc_result = self._rpc('getnetworkinfo', None)
             btc_relayfee = rpc_result.get('relayfee', btc_relayfee)
             if btc_relayfee > 0:
                 relayfee_in_sat = int(Decimal(1e8) * Decimal(btc_relayfee))
-                log.info("Using this min relay fee as tx fee floor: " +
-                    btc.fee_per_kb_to_str(relayfee_in_sat))
-                return int(max(relayfee_in_sat, random.uniform(N * float(0.8), N * float(1.2))))
             else:   # cannot get valid relayfee: fall back to 1000 sat/kbyte
-                log.info("Using this min relay fee as tx fee floor " +
-                    "(fallback): " + btc.fee_per_kb_to_str(1000))
-                return int(max(1000, random.uniform(N * float(0.8), N * float(1.2))))
+                relayfee_in_sat = 1000
```

Comment:
```python
if super().fee_per_kb_has_been_manually_set(N):
    # use the local bitcoin core relay fee as floor to avoid relay
    # problems
    rpc_result = self._rpc('getnetworkinfo', None)
    btc_relayfee = rpc_result.get('relayfee')
    if btc_relayfee:
        relayfee_in_sat = int(Decimal(1e8) * Decimal(btc_relayfee))
    else:   # cannot get valid relayfee: fall back to 1000 sat/kbyte
        relayfee_in_sat = 1000
```
Can it be like that? Why are we giving a default of -1 to `.get()` when we can use `None` directly?

Few lines below seems the same, plus I don't see why we need `retval=-1` too.
What about that:

```python
tries = 2 if N == 1 else 1
for i in range(tries):
    rpc_result = self._rpc('estimatesmartfee', [N + i])
    estimate = rpc_result.get('feerate')
    if estimate:
        retval = int(Decimal(1e8) * Decimal(estimate))
        break
else:  # cannot get valid estimate: fall back to 10000 sat/kbyte
    retval = 10000    
log.info("Using bitcoin network feerate (per kB in sats): " +\
         btc.fee_per_kb_to_str(retval))
return retval
```
Here could be nice to rename `retval` to `relayfee_in_sat`.

Reply:
> What about that:

I would agree, except: you are assuming that `None` is returned in case Core can't estimate a feerate, but I think it might be -1. I'll try to check.

Reply:
A quick regtest run showed me that your case is correct, the return is: `{'errors': ['Insufficient data or no feerate found'], 'blocks': 2}` ; I suspect I was remembering the output of `estimatefee` which preceded `estimatesmartfee`. To be honest, what we have as-is isn't so bad though, since it covers any kind of return that is invalid, correctly. Maybe simplest is: do it your way, but add a clause `and estimate > 0` just in case (it can't hurt). I'll do that.

Reply:
Done as per above in: https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/750/commits/8cdf3e0a4ac218327fb9928d81273b2ca311fc12